Mathlib/SetTheory/Ordinal/Arithmetic.lean

```diff
@@ -1058,109 +1058,204 @@ theorem comp_familyOfBFamily {o} (f : ∀ a < o, α) (g : α → β) :
+theorem sup_eq_zero_iff {ι : Type u} {f : ι → Ordinal.{max u v}} :
+    sup.{_, v} f = 0 ↔ ∀ i, f i = 0 :=
+  iSup_eq_zero_iff
+
+-- TODO: generalize universes, make sSup version.
+theorem IsNormal.iSup {f : Ordinal.{max u v} → Ordinal.{max u w}} (H : IsNormal f) {ι : Type u}
+    (g : ι → Ordinal.{max u v}) [Nonempty ι] : f (⨆ i, g i) = ⨆ i, f (g i) := by
+  apply eq_of_forall_ge_iff
+  intro a
+  rw [H.le_set' Set.univ Set.univ_nonempty g]
+  · rw [Ordinal.iSup_le_iff]
+    simp
+  · intro o
+    rw [Ordinal.iSup_le_iff]
+    simp
+
+set_option linter.deprecated false in
+@[deprecated IsNormal.iSup (since := "2024-08-27")]
 theorem IsNormal.sup {f : Ordinal.{max u v} → Ordinal.{max u w}} (H : IsNormal f) {ι : Type u}
     (g : ι → Ordinal.{max u v}) [Nonempty ι] : f (sup.{_, v} g) = sup.{_, w} (f ∘ g) :=
-  eq_of_forall_ge_iff fun a => by
-    rw [sup_le_iff]; simp only [comp]; rw [H.le_set' Set.univ Set.univ_nonempty g] <;>
-      simp [sup_le_iff]
+  H.iSup g
 
-@[simp]
+set_option linter.deprecated false in
+@[deprecated ciSup_of_empty (since := "2024-08-27")]
 theorem sup_empty {ι} [IsEmpty ι] (f : ι → Ordinal) : sup f = 0 :=
   ciSup_of_empty f
 
-@[simp]
-theorem sup_const {ι} [_hι : Nonempty ι] (o : Ordinal) : (sup fun _ : ι => o) = o :=
+set_option linter.deprecated false in
+@[deprecated ciSup_const (since := "2024-08-27")]
+theorem sup_const {ι} [_hι : Nonempty ι] (o : Ordinal) : sup (fun _ : ι => o) = o :=
   ciSup_const
 
-@[simp]
+set_option linter.deprecated false in
+@[deprecated ciSup_unique (since := "2024-08-27")]
 theorem sup_unique {ι} [Unique ι] (f : ι → Ordinal) : sup f = f default :=
   ciSup_unique
 
+protected theorem sSup_le_sSup {s t : Set Ordinal} (hst : s ⊆ t) (ht : BddAbove t) :
+    sSup s ≤ sSup t := by
+  rcases eq_empty_or_nonempty s with rfl | h
+  · simp
+  · exact csSup_le_csSup ht h hst
+
+set_option linter.deprecated false in
+@[deprecated Ordinal.sSup_le_sSup (since := "2024-08-27")]
 theorem sup_le_of_range_subset {ι ι'} {f : ι → Ordinal} {g : ι' → Ordinal}
     (h : Set.range f ⊆ Set.range g) : sup.{u, max v w} f ≤ sup.{v, max u w} g :=
-  sup_le fun i =>
-    match h (mem_range_self i) with
-    | ⟨_j, hj⟩ => hj ▸ le_sup _ _
+  Ordinal.sSup_le_sSup h (bddAbove_range.{v, max u w} _)
 
-theorem sup_eq_of_range_eq {ι ι'} {f : ι → Ordinal} {g : ι' → Ordinal}
+-- TODO: generalize or remove
+theorem iSup_eq_of_range_eq {ι ι'} {f : ι → Ordinal} {g : ι' → Ordinal}
+    (h : Set.range f = Set.range g) : iSup f = iSup g :=
+  congr_arg _ h
+
+set_option linter.deprecated false in
+@[deprecated iSup_eq_of_range_eq (since := "2024-08-27")]
+theorem sup_eq_of_range_eq {ι : Type u} {ι' : Type v}
+    {f : ι → Ordinal.{max u v w}} {g : ι' → Ordinal.{max u v w}}
     (h : Set.range f = Set.range g) : sup.{u, max v w} f = sup.{v, max u w} g :=
-  (sup_le_of_range_subset.{u, v, w} h.le).antisymm (sup_le_of_range_subset.{v, u, w} h.ge)
+  Ordinal.iSup_eq_of_range_eq h
```

Comment:
Same here. Diff is a mess. Please make sure the statement of `sup_eq_of_range_eq` is not part of the diff (the proof can, however)